### PR TITLE
Add flake8 to CI/CD

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
After the big whitespace cleanup, I figured it'd be a good idea to avoid adding that again. One way of doing that is to use flake8 in the CI/CD.

#### Description of change
This PR:
- Adds flake8 to the dev requirements.
- Runs flake8 as a part of tox.
- Adds flake8 settings that makes it pass for the existing code base.
- Removes trailing new lines for a set of files.

#### Steps to Test
Add whitespace to an arbitrary Python file in the `<repo root>/arelle/` directory and run `flake8 arelle`. You should be seeing this:
```
(.venv) vscode ➜ /workspaces/Arelle (master) $ flake8 arelle
arelle/PackageManager.py:37:41: W291 trailing whitespace
```

My VS code settings cleans up white space automatically so I had to use a text editor to add the white space before testing.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: setup.cfg [hunk 1]
```diff
@@ -2,3 +2,159 @@
...
+# E731 do not assign a lambda expression, use a def
+# E741 ambiguous variable name
+
+# F401 imported but unused
+# F402 import 'xxx' from line 15 shadowed by loop variable
+# F403 'from xxx import *' used; unable to detect undefined names
+# F405 xxx may be undefined, or defined from star imports: xxx
+# F504 '...' % ... has unused named argument(s)
+# F523 '...'.format(...) has unused arguments at position(s): n
+# F524 '...'.format(...) is missing argument(s) for placeholder(s): n
```
Comment (ggravlingen) [inline setup.cfg:75]: I found one of these in the wild when I was adding type hints. `F524` could be a possible bug?

File: setup.cfg [hunk 2]
```diff
@@ -2,3 +2,159 @@
...
+# F811 redefinition of unused 'xxx' from line 31
+# F841 local variable name is assigned to but never used
+
+# W191 indentation contains tabs
+# W503 line break before binary operator
+# W504 line break after binary operator
+# W605 invalid escape sequence '\d'
+
+ignore =
+    E101,
```
Comment (sagesmith-wf) [inline setup.cfg:88]: Can the comments go inline with the error code here? I worry that if these change the comments aren't going to get updated in the future

File: setup.cfg [hunk 3]
```diff
@@ -2,3 +2,159 @@
...
+    E241,
+    E251,
+    E252,
+    E261,
+    E262,
+    E265,
+    E266,
+    E272,
+    E271,
+    E301,
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply setup.cfg:124]: That's odd. What version of Python are you running?

File: setup.cfg (hunk 2 above)
Comment (ggravlingen) [reply setup.cfg:88]: You mean putting the comment above the ignored code?

Comment (sagesmith-wf) [reply setup.cfg:88]: Was thinking maybe something like:

``` python
W605, # invalid escape sequence '\d'
```

Comment (ggravlingen) [reply setup.cfg:88]: That would work

File: setup.cfg [hunk 4]
```diff
@@ -2,3 +2,159 @@
 source-dir = apidocs
 build-dir  = apidocs/_build
 all_files  = 1
+
+[flake8]
+exclude = .venv,.git,.tox,docs,venv,bin,lib,deps,build
+doctests = True
+noqa-require-code = True
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply setup.cfg:9]: Good catch

File: arelle/plugin/validate/XFsyntax/xf.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def __init__(
         )
 
     @tatsumasu()
-    def _module_(self):  # noqa
```
Comment (ggravlingen) [inline arelle/plugin/validate/XFsyntax/xf.py:83]: All of these noqa yielded errors like this:
`arelle/plugin/validate/XFsyntax/xf.py:1907:28: NQA101 "# noqa" has no violations`

**Review by sagesmith-wf — DISMISSED** [review]

**Comment by ggravlingen** [comment]
Please let me rebase before you merge. The commit history became a bit messy here.

**Comment by ggravlingen** [comment]
> I don't think you need to reset. You can use `git rebase -i 057c5125~` to clean up the commit history for this PR.

Thanks! I think I got that right except for the commit message.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**